### ISSUE TITLE
Thinking Reward Dataset

### ISSUE THREAD

**Issue by hxgr4ce** [kxfan2002/SophiaVL-R1#11]
Hi! I noticed that in the Thinking Reward Dataset,`TextOCR/`, `TextCaps/`, `HME100k/`, and `EXAMS-V/` each only have one single image in them, even though when I look through `multimodal_part.json`, there are around 25k samples that are from these sources. For example, ~5k samples (so all samples) with "TextOCR" as the data source are all linked to the single image `./OCR/TextOCR/2fb8fe3ffedbf815.png`

I've noticed a couple other discrepancies in samples from other data sources as well (e.g. all 214 "What is the capital of this state?" questions in ScienceQA seem to be mapped to `./Knowledge/ScienceQA/images/12780.png`). I notice this even for higher scoring samples (e.g. in the ScienceQA case, 32 of the thinking processes are scored as a 1). 

Was this uploaded by mistake? Is there a newer version of the dataset? Thanks!

**Comment by kxfan2002** [comment]
> Hi! I noticed that in the Thinking Reward Dataset,`TextOCR/`, `TextCaps/`, `HME100k/`, and `EXAMS-V/` each only have one single image in them, even though when I look through `multimodal_part.json`, there are around 25k samples that are from these sources. For example, ~5k samples (so all samples) with "TextOCR" as the data source are all linked to the single image `./OCR/TextOCR/2fb8fe3ffedbf815.png`
> 
> I've noticed a couple other discrepancies in samples from other data sources as well: e.g. all 214 "What is the capital of this state?" questions in ScienceQA seem to be mapped to `./Knowledge/ScienceQA/images/12780.png`. I notice this even for higher scoring samples: e.g. in the ScienceQA case, 32 of the thinking processes are scored as a 1. I can see how this could result from many rounds of GRPO, however the question options vary, even though the image remains `./Knowledge/ScienceQA/images/12780.png`.
> 
> Was this uploaded by mistake? Is there a newer version of the dataset? Thanks!

Hi Grace,

Thanks again for your careful observation — I really appreciate it.

You are absolutely right. There is a mistake in my data construction 😢. I did not record the answer and image path properly, and instead used `'question'` as the matching key. This obviously causes the issue you pointed out. Unfortunately, I missed it during initial checks — my fault.

I will re-run the generation process and prepare a new batch of data (which might differ from the current one I uploaded, this process takes about 15 days due to computational limitation). Alternatively, if you prefer to regenerate this part of data yourself and upload it (which I would greatly appreciate), you can follow these steps:

1. Generate responses from the base model using the GRPO training code (`EasyR1`). Print the question, answer, ground truth and image path. Extract question pairs from the log.
2. Score these responses using `Qwen2.5VL-72B-Instruct`. I use  Here is a code snippet for your reference:
```python
def get_response(question, answer, images, answer_score):
    global omit_count
    if images != "":
        question = f"<image> {question}"
    prompt = f"""
        You are a Large Reasoning model with advanced reasoning capabilities. Your goal is to judge a reward process and give a score between 0 and 1. You should focus on whether the reasoning process is good rather than whether the final answer is correct.
        ### Evaluation Criteria:
        - Logical Soundness: Does each step follow logically from the previous one?
        - Correct Reasoning: Are the methods and steps used appropriate and valid? Are the facts and lemmas correctly stated and applied?
        - Error Identification: Are there any logical fallacies, unsupported assumptions, or incorrect steps?
        - Language Consistency: Is the reasoning process conducted in a single, consistent language without mixing different languages or messy characters?
        - Redundancy: Is the reasoning concise, without unnecessary repetition or extraneous steps?

        ### Input:
        - Question: {question}
        - Reasoning Process: {answer}

        ### Output Format:
        Provide a single score from {0, 0.1, 0.2, ..., 1.0} based on the reasoning quality, where:
        - 0: Completely flawed reasoning
        - 1: Perfectly sound reasoning
        - Intermediate values (e.g., 0.3, 0.7) should reflect partial correctness or minor errors.

        Your answer should be put in \\box{{your_score}}. Be strict — reward good reasoning and penalize poor reasoning. Output the score only.
    """
    prompt = sanitize_prompt(prompt)
    messages = [
        {"role": "system", "content": "You are a helpful assistant."},
        {
            "role": "user",
            "content": [{"type": "text", "text": prompt}],
        },
    ]
    if images != "":
        img_path = os.path.join(base_path, images)
        img_enc = encode_image(img_path)
        image_url_entry = {
            "type": "image_url",
            "image_url": {
                "url": f"data:image/{images.split('.')[-1]};base64,{img_enc}",
            },
        }
        messages[1]["content"].append(image_url_entry)

    retries = 0
    while retries < max_retries:
        try:
            response = client.chat.completions.create(
                model="Qwen2.5-VL-72B-Instruct",
                messages=messages,
            )
            break
        except InternalServerError:
            omit_count += 1
            print(f"[Omit {omit_count}] prompt = {question}")
            retries += 1
            wait_time = 2 ** retries
            print(f"Server error. Retrying... ({retries}/{max_retries}), waiting {wait_time}s")
            time.sleep(wait_time)
            continue
    else:
        return None

    result = response.choices[0].message.content
    score = extract_score(result)
    if score is not None:
        return {
            "question": question,
            "answer": answer,
            "images": images,
            "answer_score": answer_score,
            "process_score": score,
        }
    return None
```
